src/Dict/Extra.elm

```diff
@@ -380,3 +386,59 @@ find predicate dict =
         )
         Nothing
         dict
+
+
+{-| Combine two dictionaries. If there is a collision, a combining function is
+used to combine the two values.
+
+    import Dict
+
+    unionWith (+)
+        (Dict.fromList [ ( "a", 2 ), ( "b", 3 ) ])
+        (Dict.fromList [ ( "b", 4 ), ( "c", 1 ) ])
+        --> Dict.fromList [ ( "a", 2 ), ( "b", 7 ), ( "c", 1 ) ]
+
+    unionWith (++)
+        (Dict.fromList [ ( 1, "123" ), ( 2, "abc" ) ])
+        (Dict.fromList [ ( 2, "def" ), ( 3, "xyz" ) ])
+        --> Dict.fromList [ ( 1, "123" ), ( 2, "abcdef" ), ( 3, "xyz" ) ]
+
+Note that, like `Dict.union`, it is more efficient to have the larger `Dict` as
+the second argument, i.e. when possible, you should use `unionWith f new old`,
+if `old` has more keys than `new`.
```

Comment:
I’m not sure how you feel about this w.r.t. beginner-friendliness, but I would appreciate this sort of comment to include the Big O complexity here, i.e. O(n _log_ m), where n is the first arg and m is the second.

Reply:
Ok, we might update that later.